third_party/move/move-compiler-v2/tests/variable-coalescing/branch_1.exp

```diff
@@ -130,16 +130,15 @@ foo(Arg0: bool, Arg1: u64): u64 /* def_idx: 0 */ {
 L2:	loc0: u64
 B0:
 	0: MoveLoc[0](Arg0: bool)
-	1: BrFalse(5)
+	1: BrFalse(4)
 B1:
 	2: LdU64(0)
 	3: StLoc[1](Arg1: u64)
-	4: Branch(5)
 B2:
-	5: MoveLoc[1](Arg1: u64)
-	6: LdU64(1)
-	7: Add
-	8: Ret
+	4: MoveLoc[1](Arg1: u64)
+	5: LdU64(1)
+	6: Add
+	7: Ret
```

Comment:
I have checked all the outputs to see that do better or equal code gen in all cases (at least in terms of number of instructions). You probably already had this planned, but after addressing all comments and getting approvals, instead of auto-merging, please double check that there are no regressions due to changes, and re-measure the performance benefit on aptos-framework.

Reply:
I remeasured the performance improvement, which is 2.8%. If we disable the flush writes optimization, that'll be 3.4%.

Reply:
That is interesting. The new flush writes is even better, I wonder if that affects these numbers much. Anyway, we can remeasure after both land.